sdk/ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -478,13 +477,13 @@ object WebSocketService {
       }
 
       private def prepareFilters(
-          resolved: NonEmpty[Set[domain.ContractTypeId.Resolved]],
+          resolved: NonEmpty[Set[_ <: ContractTypeRef.Resolved]],
           queryExpr: Map[String, JsValue],
           lookupType: ValuePredicate.TypeLookup,
       ): CompiledQueries =
-        resolved.toSeq.map { tid =>
-          val vp = ValuePredicate.fromTemplateJsObject(queryExpr, tid, lookupType)
-          (tid, (vp, vp.toFunPredicate))
+        resolved.flatMap { tid =>
+          val vp = ValuePredicate.fromTemplateJsObject(queryExpr, tid.latestPkgId, lookupType)
```

Comment:
We use the latest package id per contract type ref, to interpret the query object.